gallery_dl/extractor/danbooru.py

```diff
@@ -126,7 +127,7 @@ def _pagination(self, endpoint, params, pagenum=False):
                 posts = posts["posts"]
             yield from posts
 
-            if len(posts) < self.per_page:
+            if self.strategy == "length" and len(posts) < self.per_page:
```

Comment:
I'd suggest introducing a new `limit` variable at the start of `_pagination()` and setting it to `self.per_page` or `1` depending on `self.strategy` and using it like `if len(posts) < limit: return`. Seems cleaner to me.

Maybe this option should better be called something like `limit` with `"auto"` (`self.per_page` or `self.per_page // 2`?) being the default and otherwise the number of returned posts when to stop?

Reply:
> Maybe this option should better be called something like `limit` with `"auto"` (`self.per_page` or `self.per_page // 2`?) being the default and otherwise the number of returned posts when to stop?

Currently gallery-dl does not actually check if the value is `"auto"`. Should I only allow `int` and `"auto"` and raise an error if an invalid type is specified?

Reply:
I'd check if the value is `"auto"` or `None` and use `self.per_page` in this case, and assume it is `int` otherwise and apply `text.parse_int` to it just to make sure. At least that is what I've usually done in such cases.

Your implementation from https://github.com/mikf/gallery-dl/pull/3413/commits/dd4a4a3fa622d3cf6030afb19e7474bd0feafb79 is fine as well.